tests/acceptance/fetch_test.py

```diff
@@ -252,3 +275,41 @@ def test_fido_request_no_timeout_when_header_value_not_list(tcp_nodelay):
         },
         tcp_nodelay=tcp_nodelay,
     ).wait(timeout=5)
+
+
+def test_fido_request_gzip_enabled(server_url):
+    expected_body = b'hello world'
+
+    # Ensure invalid gzipped responses raise a
+    # GzipDecompressionError exception.
+    with pytest.raises(GzipDecompressionError):
+        response = fido.fetch(
+            server_url + ECHO_URL,
+            headers={'Content-Encoding': 'gzip'},
```

Comment:
The idea with this test is to trick the client into decompressing the text response. Since we're echoing the request headers and body, the response content-encoding header tells the client that gzip is the encoding type. 

This test could definitely use better documentation!